tests/database/test_unittestdb.py

```diff
@@ -66,31 +84,42 @@ def test_init(
         with expectation:
             server_url = request.config.getoption("server")
             src_path = src if src.is_absolute() else data_dir / src
-            db_key = name if name else src.name
-            self.dbs[db_key] = UnitTestDB(server_url, src_path, name)
-            # Check that the database has been created correctly
-            assert self.dbs[db_key], "UnitTestDB should not be empty"
-            assert self.dbs[db_key].dbc, "UnitTestDB's database connection should not be empty"
-            # Check that the database has been loaded correctly from the dump files
-            result = self.dbs[db_key].dbc.execute("SELECT * FROM gibberish")
-            assert len(result.fetchall()) == 6, "Unexpected number of rows found in 'gibberish' table"
-
-    @pytest.mark.dependency(depends=["test_init"], scope="class")
+            with UnitTestDBContext(server_url, dump_dir=src_path, name=name, tmp_path=tmp_path) as db:
+                # Check that the database has been created correctly
+                assert db, "UnitTestDB should not be empty"
+                assert db.dbc, "UnitTestDB's database connection should not be empty"
+                # Check that the database has been loaded correctly from the dump files
+
+                with db.dbc.test_session_scope() as session:
+                    result = session.execute(text("SELECT * FROM gibberish"))
+                    assert len(result.fetchall()) == 6, "Unexpected number of rows found in 'gibberish' table"
```

Comment:
There is a small overhead when using the `test_session_scope()`, so maybe we could use `connection` instead?